### ISSUE TITLE
Ability to specify Dockerfile (--file)

### ISSUE THREAD

**Issue by gvangool** [docker/hub-feedback#292]
Is it possible to add the ability to specify Dockerfile (the `--file` option on the cli)?

Currently I build with a Makefile, but since we're moving to production we want to switch to automated builds:

``` make
build-httpd:
        docker build -t gvangool/repo:httpd -f _build/Dockerfile_httpd .

build-fax:
        docker build -t gvangool/repo:fax -f _build/Dockerfile_fax .
```

**Comment by ChristopherBiscardi** [comment]
To be clear, you are asking for the ability to separately specify the build context and the dockerfile in Automated Builds?

**Comment by gvangool** [comment]
Yes indeed. In my case the context would be `/`, while the dockerfile is `_build/Dockerfile_httpd`. At this moment you can only specify the context.

**Comment by ChristopherBiscardi** [comment]
It is not possible to do this in Automated Builds today. Marking as feature request.

**Comment by stooie90** [comment]
+1 for this. I use that style for separating out my dev and production environments.

**Comment by jakirkham** [comment]
+1 also would really like to see the option to specify build context and `Dockerfile` separately. This is incredibly useful if several docker images rely on many of the same components and hence are in the same repo, but at a higher directory level than the `Dockerfile`s.

**Comment by geoah** [comment]
+1 Currently have a repo that produces multiple containers. Would be very useful to allow container repos to specify dockerfiles and not just their context paths.

**Comment by isntall** [comment]
+1 I have pretty much the same use case.

**Comment by akoi** [comment]
+1 Would be great to be able to keep the dockerfiles out of the project/repo root.

**Comment by glinton** [comment]
+1

**Comment by galenp** [comment]
+1

**Comment by agilgur5** [comment]
+1

**Comment by gillisd** [comment]
+1

**Comment by inanimatt** [comment]
+1 from me too, though I'm making do right now by symlinking the various Dockerfiles to my project root.

**Comment by d7h** [comment]
+1

**Comment by janeczku** [comment]
+1 
Also "Dockerfile Location" is very misleading because it doesn't just fetch the Dockerfile from that location but also changes the context!

**Comment by RongxinZhang** [comment]
+1. Agreed I think that most user have multiple dockerfiles, for development, staging and production.

**Comment by Primer42** [comment]
:+1:

**Comment by avif** [comment]
+1

**Comment by snopoke** [comment]
+1

**Comment by rsanaie** [comment]
+1

**Comment by guruvan** [comment]
+1 as of the above I thought specifying a dockerfile was already available, but it would turn out that on some projects we've had to move off automated builds (for this and other reasons) and most of the other projects we've retooled to work from separate branches which isn't always appropriate. 

I'd consider it a bonus if I could specify a build order for some project, and build multiple dockerfiles in succession.

**Comment by Richard-Mathie** [comment]
Arrgh +1!
So frustrating,
I guess as a work around you could maintain different branches for each "Different" Dockerfile. But it suck so much.

**Comment by rrarunan** [comment]
We have a plan in place and will provide an update as soon as we can.
